### ISSUE TITLE
Centralized logging solution

### ISSUE THREAD

**Issue by jrochkind** [sciencehistory/scihist_digicoll#569]
right now we might want to look at Rails app logs on web server and jobs server. That's just two servers, and different kinds of activity is logged on both. 

If we horizontally scale *either* web or jobs, looking at logs is going to get a lot harder. 

We might want to investigate some centralized logging solution, where logs from different  servers are all aggregated. Some such solutions also give you a web UI for searching through them etc. 

This could be open source software we run in-house, or it could be a cloud solution. There are a variety of commercial cloud offerings in this space. I think Stanford has one they like, we should check with them to at least take a look at that one.

**Comment by sanfordd** [comment]
This also links to our earlier talk about privacy concerns in #519  with R&R requests. If we use an external log aggregator then we'll want to figure out how we should handle any sensitive information before it gets passed out.

**Comment by sanfordd** [comment]
We want to ask is what sort of centralized logging solution we're looking for. Broadly speaking we have 3 types of logs on our digital collection servers
1. Application logs 
2. Application related logs (i.e. Solr, Redis, Postgres)
3. General system logs

If we're scaling out we can assume that we definitely want 1 and probably 2. Including general system logs is a broader question, if we include there's greater ease in troubleshooting problems that exist outside the application but it will increase the scale of logs we'd be dealing with. Using web production as an example, our application logs for a month of data are approximately 800 MB, system logs (which vary a bit more in time range) are approximately 1.2 GB worth of data.

**Comment by jrochkind** [comment]
I've been using papertrail with heroku (the free tier), as it's one of the plugins available for logging -- on heroku you kind of need *some* plugin to have logs be very accessible at all. I am liking papertrail! (I started out with a different heroku add-on I did not like as much). 

Papertrail can be used without heroku too, although the pricing tiers are somewhat different on/off heroku (not totally sure which are 'better').  It also would require some changes to machines (via ansible presumably) to include necessary setup for capturing logging for papertrail. 

https://www.papertrail.com/plans
https://documentation.solarwinds.com/en/Success_Center/papertrail/Content/kb/configuration/configuring-centralized-logging-from-ruby-on-rails-apps.htm
